pennylane/data/dataset.py

```diff
@@ -158,11 +166,11 @@ def attrs(self):
         return {k: v for k, v in vars(self).items() if k[0] != "_"}
 
     @staticmethod
-    def _read_file(filepath):
-        """Read data from a saved file.
+    def get_file_contents(filepath):
```

Comment:
Does it require a changelog entry? Also, should we make `_write_file` public too?

Reply:
I'll add a changelog entry. IMO we should not make `_write_file` public, because any dataset written to file with `write()` can be read into another dataset nicely using `read()`, whereas arbitrary data written by `_write_file` may or may not be readable by `read()`.

I'd prefer to encourage users to use `write()` and `read()` since they cover the normal, desired use cases.